antlr_ast.py

```diff
@@ -61,9 +62,15 @@ class AstNode(AST):
     _rules = []
 
     def __init__(self, _ctx=None, **kwargs):
-        # TODO: ensure key is in _fields?
+        # This makes instances compatible with tree walking
+        self.__fields = self._fields
```

Comment:
Btw `__` implies special behaviour in python. It makes something "private", not only by convention, but also by mangling the real method name. `AstNode()` in this case won't have the attribute `__fields` but the attribute `_AstNode__fields`. Internally, you can refer to it with `self.__fields` of course. Subclasses would have to use the mangled attribute name.

I'm just mentioning this because I wasn't sure whether this is the behaviour you need here.